### PR TITLE
Ceph volume purge container

### PR DESCRIPTION
**Pull request by leseb**

### CODE REVIEW

**Comment by leseb** [comment]
Works on top of https://github.com/ceph/ceph-ansible/pull/2866 so need a rebase when https://github.com/ceph/ceph-ansible/pull/2866 gets merged.

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by leseb** [comment]
We need an lvm purge container scenario to validate this

File: infrastructure-playbooks/purge-docker-cluster.yml [hunk 1]
```diff
@@ -271,117 +271,150 @@
...
+          journal: "{{ item.journal|default(omit) }}"
+          journal_vg: "{{ item.journal_vg|default(omit) }}"
+          db: "{{ item.db|default(omit) }}"
+          db_vg: "{{ item.db_vg|default(omit) }}"
+          wal: "{{ item.wal|default(omit) }}"
+          wal_vg: "{{ item.wal_vg|default(omit) }}"
+          action: "zap"
+        environment:
+          CEPH_VOLUME_DEBUG: 1
+          CEPH_CONTAINER_IMAGE: "docker.io/ceph/daemon:latest-luminous"
```
Comment (leseb) [inline infrastructure-playbooks/purge-docker-cluster.yml]: do not hardcode the image name

File: infrastructure-playbooks/purge-docker-cluster.yml [hunk 2]
```diff
@@ -271,117 +271,150 @@
...
+        with_items: "{{ lvm_volumes }}"
+
+      - name: zap and destroy osds created by ceph-volume with devices
+        ceph_volume:
+          data: "{{ item }}"
+          action: "zap"
+        environment:
+          CEPH_VOLUME_DEBUG: 1
+          CEPH_CONTAINER_IMAGE: "docker.io/ceph/daemon:latest-luminous"
```
Comment (leseb) [inline infrastructure-playbooks/purge-docker-cluster.yml]: ditto

**Comment by leseb** [comment]
Needs https://github.com/ceph/ceph-ansible/pull/3395/files

**Comment by leseb** [comment]
jenkins test dev-purge_lvm_osds_container

**Comment by leseb** [comment]
jenkins test dev-purge_cluster_non_container

**Comment by leseb** [comment]
jenkins test dev-purge_cluster_container

**Comment by leseb** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1653307

**Comment by leseb** [comment]
jenkins test dev-purge_cluster_container

**Comment by leseb** [comment]
Needs https://github.com/ceph/ceph-ansible/pull/3412

**Comment by guits** [comment]
jenkins test dev-purge_cluster_container

**Comment by guits** [comment]
jenkins test dev-purge_cluster_container

**Comment by guits** [comment]
jenkins test dev-purge_cluster_container

**Comment by guits** [comment]
failure because of ceph/ceph#25425

**Comment by guits** [comment]
since mergify won't merge because of the unrelated failure seen in lvm-batch scenario, I'm merging this myself anyway.

**Pull request merged**